kafka-consumer/src/main/java/com/demo/kafka/controller/KafkaConsumerExample.java

```diff
@@ -0,0 +1,45 @@
+package com.demo.kafka.controller;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Arrays;
+import java.util.Properties;
+
+@RestController
+@RequestMapping("/kafkaDemo")
+public class KafkaConsumerExample {
+
+    @GetMapping("/publish")
+    public void produce() {
+
+        String topic = "demo";
```

Comment:
define application property for topic and pass it to the consumer during initialization.